netutils/netlib/netlib_iptables.c

```diff
@@ -0,0 +1,570 @@
+  memmove(base + insert_point + entry->next_offset, base + insert_point,
+          repl->size - insert_point);
+  memcpy(base + insert_point, entry, entry->next_offset);
+
+  /* Adjust metadata. */
+
+  repl->num_entries++;
+  repl->size += entry->next_offset;
+
+  /* Adjust hook_entry and underflow. */
+
+  repl->underflow[hook++] += entry->next_offset;
+  for (; hook < NF_INET_NUMHOOKS; hook++)
+    {
+      if (repl->valid_hooks & (1 << hook))
+        {
+          repl->hook_entry[hook] += entry->next_offset;
+          repl->underflow[hook] += entry->next_offset;
+        }
+    }
+
+  *replace = repl;
+  return OK;
+}
+
+/****************************************************************************
+ * Name: netlib_ipt_delete_internal
+ *
+ * Description:
+ *   Delete an entry from config.
+ *
+ * Input Parameters:
+ *   repl   - The config (to set into kernel later).
+ *   entry  - The entry to remove, should be in repl.
+ *   hook   - The hook of the entry.
+ *
+ ****************************************************************************/
+
+static void netlib_ipt_delete_internal(FAR struct ipt_replace *repl,
+                                       FAR struct ipt_entry *entry,
+                                       enum nf_inet_hooks hook)
+{
+  unsigned int delete_len = entry->next_offset;
+
+  /* Adjust metadata. */
+
+  repl->num_entries--;
+  repl->size -= delete_len;
+
+  /* Remove entry from entry table. */
+
+  memmove((FAR uint8_t *)entry, (FAR uint8_t *)entry + delete_len,
+          repl->size - ((uintptr_t)entry - (uintptr_t)repl->entries));
+
+  /* Adjust hook_entry and underflow. */
+
+  repl->underflow[hook++] -= delete_len;
+  for (; hook < NF_INET_NUMHOOKS; hook++)
+    {
+      if (repl->valid_hooks & (1 << hook))
+        {
+          repl->hook_entry[hook] -= delete_len;
+          repl->underflow[hook] -= delete_len;
+        }
+    }
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: netlib_ipt_prepare
+ *
+ * Description:
+ *   Read current config from kernel space.
+ *
+ * Input Parameters:
+ *   table  - The table name to read from.
+ *
+ * Returned Value:
+ *   The pointer to the config, or NULL if failed.
+ *   Caller must free it after use.
+ *
+ ****************************************************************************/
+
+FAR struct ipt_replace *netlib_ipt_prepare(FAR const char *table)
+{
+  struct ipt_getinfo info;
+  FAR struct ipt_get_entries *entries;
+  FAR struct ipt_replace *repl = NULL;
+  socklen_t len;
+  int sockfd;
+
+  if (table == NULL)
+    {
+      return NULL;
+    }
+
+  sockfd = socket(NET_SOCK_FAMILY, NET_SOCK_TYPE, NET_SOCK_PROTOCOL);
+  if (sockfd < 0)
+    {
+      fprintf(stderr, "Failed to create socket %d!\n", get_errno());
```

Comment:
`get_errno` is a KERNEL level API, so I suppose just `errno` should be used in apps

Reply:
Done.